cloud/services/domains.go

```diff
@@ -97,37 +103,40 @@ func DeleteIPFromDNS(ctx context.Context, mscope *scope.MachineScope) error {
 	return nil
 }
 
-// GetMachinePublicIP gets the machines public IP
-func GetMachinePublicIP(ctx context.Context, mscope *scope.MachineScope) (string, error) {
+// GetMachinePublicIPs gets the machines public IP
+func GetMachinePublicIPs(ctx context.Context, mscope *scope.MachineScope) (map[string]string, error) {
```

Comment:
Since we're only calling this from this file, this should probably be fine to unexport?